docs/self-hosting.md

```diff
@@ -0,0 +1,72 @@
+# Self-Hosting Guide for Discuit
+
+Welcome to the self-hosting guide for Discuit! Here, you'll find step-by-step instructions on how to set up and run Discuit on your own server.
+
+## Prerequisites
+
+Before you begin, make sure you have the following installed on your system:
+
+- [Go (1.21 or higher)](https://go.dev/)
+- [MariaDB](https://mariadb.org/)
+- [Redis](https://redis.io/) or [Valkey](https://valkey.io/) if you prefer a proper open-source alternative fork
```

Comment:
Have you tested with Valkey to make sure it works? If not, then I'd rather not suggest it unless we know for sure that it works.

Reply:
I'll remove it, currently, it's just a fork of Redis due to the license change, however, the API may change in the future, though it may be possible in the future to switch discuit over to use that.